docs/developer/architecture/core/logging-service.asciidoc

```diff
@@ -0,0 +1,545 @@
+[[logging-service]]
+== Logging service
+Allows a plugin to provide status and diagnostic information.
+
+[source,typescript]
+----
+import type { PluginInitializerContext, CoreSetup, Plugin, Logger } from 'kibana/server';
+
+export class MyPlugin implements Plugin {
+  private readonly logger: Logger;
+
+  constructor(initializerContext: PluginInitializerContext) {
+    this.logger = initializerContext.logger.get();
+  }
+
+  public setup(core: CoreSetup) {
+    try {
+      this.logger.debug('doing something...');
+      // …
+    } catch (e) {
+      this.logger.error('failed doing something...');
+    }
+  }
+}
+----
+
+The way logging works in {kib} is inspired by `log4j 2` logging framework used by {ref-bare}/current/logging.html[Elasticsearch].
+The main idea is to have consistent logging behaviour (configuration, log format etc.) across the entire Elastic Stack where possible.
```

Comment:
nit: The parenthetical is probably not necessary here.

@lcawl What's our protocol for US English vs UK English spellings in the docs, do we prefer one or the other? Specifically referring to "behaviour" below.

```suggestion
The main idea is to have consistent logging behaviour across the entire Elastic Stack where possible.
```